### PR TITLE
feat(DMS): add consumer group resource

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add resource kafka consumer group
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dms/ TESTARGS='-run TestAccDmsKafkaConsumerGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDmsKafkaConsumerGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsKafkaConsumerGroup_basic
=== PAUSE TestAccDmsKafkaConsumerGroup_basic
=== CONT  TestAccDmsKafkaConsumerGroup_basic
--- PASS: TestAccDmsKafkaConsumerGroup_basic (815.76s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       815.849s

```

### CODE REVIEW

File: docs/resources/dms_kafka_consumer_group.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+## Import
+
+The kafka consumer group can be imported using the kafka instance ID and group name separated by a slash, e.g.
+
+```
+$ terraform import huaweicloud_dms_kafka_consumer_group.test 8d3c7938-dc47-4937-a30f-c80de381c5e3/group_1
```
Comment (houpeng80) [inline docs/resources/dms_kafka_consumer_group.md]:
```suggestion
$ terraform import huaweicloud_dms_kafka_consumer_group.test <instance_id>/<name>
```

File: docs/resources/dms_kafka_consumer_group.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+## Import
+
+The kafka consumer group can be imported using the kafka instance ID and group name separated by a slash, e.g.
```
Comment (houpeng80) [inline docs/resources/dms_kafka_consumer_group.md]:
```suggestion
The kafka consumer group can be imported using the kafka `instance_id` and `name` separated by a slash, e.g.
```

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+func getDmsKafkaConsumerGroupResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getKafkaConsumerGroup: query DMS kafka consumer group
+	var (
+		getKafkaConsumerGroupHttpUrl = "v2/{project_id}/instances/{instance_id}/groups/{group}"
+		getKafkaConsumerGroupProduct = "dmsv2"
+	)
+	getKafkaConsumerGroupClient, err := cfg.NewServiceClient(getKafkaConsumerGroupProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating DmsKafkaConsumerGroup Client: %s", err)
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating DMS Client: %s", err)
```
and fix other places

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	}
+
+	getKafkaConsumerGroupRespBody, err := utils.FlattenResponse(getKafkaConsumerGroupResp)
+	if err != nil {
+		return nil, err
+	}
+	groupJson := utils.PathSearch("group", getKafkaConsumerGroupRespBody, nil)
+	groupState := utils.PathSearch("state", groupJson, nil)
+	if groupState == "DEAD" {
+		return nil, fmt.Errorf("error retrieving DmsKafkaConsumerGroup")
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go]:
```suggestion
		return nil, fmt.Errorf("error retrieving DMS Kafka consumer group")
```
and fix other places

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the name of the consumer group.`,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[A-Za-z|%-_0-9]*$`),
+						"An instance name starts with a letter and can contain only letters, digits,"+
+							"vertical lines(|), percent sign(%), underscores (_), and hyphens (-)"),
+					validation.StringLenBetween(3, 64),
+				),
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]: delete the validataion func. it is not used now

File: docs/resources/dms_kafka_consumer_group.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  Changing this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies the name of the consumer group.
+
+  Changing this parameter will create a new resource.
+
+* `description` - (Optional, String) Specifies the description of the consumer group.
+
+* `state` - (Optional, String) Specifies the state of the consumer group.
+  Value options: **DEAD**, **EMPTY**, **PreparingRebalance**, **CompletingRebalance**, **Stable**.
```
Comment (houpeng80) [inline docs/resources/dms_kafka_consumer_group.md]:
```suggestion
* `state` - Indicates the state of the consumer group.
  Value options: **DEAD**, **EMPTY**, **PreparingRebalance**, **CompletingRebalance**, **Stable**.
```
and fix other attributes of coordinator_id, lag and created_at

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getDmsKafkaConsumerGroupResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getKafkaConsumerGroup: query DMS kafka consumer group
+	var (
+		getKafkaConsumerGroupHttpUrl = "v2/{project_id}/instances/{instance_id}/groups/{group}"
+		getKafkaConsumerGroupProduct = "dmsv2"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go]:
```suggestion
		getKafkaConsumerGroupProduct = "dms"
```
only the endpoint of the client is used in this resource, so the version is not needed,  and fix other places

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+		return diag.Errorf("error creating DmsKafkaConsumerGroup: %s", createErr)
+	}
+
+	d.SetId(instanceID + "/" + d.Get("name").(string))
+	return resourceDmsKafkaConsumerGroupRead(ctx, d, meta)
+}
+
+func buildCreateKafkaConsumerGroupBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"group_name": utils.ValueIngoreEmpty(d.Get("name")),
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]:
```suggestion
		"group_name": d.Get("name"),
```
the param of name is required, so it is not needed to judge whethere the value should ignore

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+	var (
+		deleteKafkaConsumerGroupHttpUrl = "v2/{project_id}/instances/{instance_id}/groups/batch-delete"
+		deleteKafkaConsumerGroupProduct = "dmsv2"
+	)
+	deleteKafkaConsumerGroupClient, err := cfg.NewServiceClient(deleteKafkaConsumerGroupProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DmsKafkaConsumerGroup Client: %s", err)
+	}
+
+	parts := strings.SplitN(d.Id(), "/", 2)
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]:
```suggestion
	parts := strings.Split(d.Id(), "/")
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+	_, err = deleteKafkaConsumerGroupClient.Request("POST", deleteKafkaConsumerGroupPath, &deleteKafkaConsumerGroupOpt)
+	if err != nil {
+		return diag.Errorf("error deleting DmsKafkaConsumerGroup: %s", err)
+	}
+	return resourceDmsKafkaConsumerGroupRead(ctx, d, meta)
+}
+
+func buildBatchDeleteKafkaConsumerGroupBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"group_ids": buildBatchDeleteResourcesRequestBodyOpts(utils.ValueIngoreEmpty(d.Get("name"))),
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]:
```suggestion
		"group_ids": []strintg{d.Get("name").(string)},
```

File: docs/resources/dms_kafka_consumer_group.md (hunk 1 above)
Comment (346010762) [reply docs/resources/dms_kafka_consumer_group.md]: done

File: docs/resources/dms_kafka_consumer_group.md (hunk 2 above)
Comment (346010762) [reply docs/resources/dms_kafka_consumer_group.md]: done

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go (hunk 3 above)
Comment (346010762) [reply huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go]: done

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go (hunk 4 above)
Comment (346010762) [reply huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go (hunk 5 above)
Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]: done

File: docs/resources/dms_kafka_consumer_group.md (hunk 6 above)
Comment (346010762) [reply docs/resources/dms_kafka_consumer_group.md]: done

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go (hunk 7 above)
Comment (346010762) [reply huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go (hunk 8 above)
Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go (hunk 9 above)
Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go (hunk 10 above)
Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]: done

**Pull request closed without merging**